optimum/habana/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -571,3 +577,131 @@ def forward(
             attentions=transformer_outputs.attentions,
             cross_attentions=transformer_outputs.cross_attentions,
         )
+
+
+class GaudiGPT2DoubleHeadsModel(GPT2DoubleHeadsModel):
```

Comment:
just curious: this class is added because its needed for contrastive search?

Reply:
I added it to fix the pytest for contrastive search with low memory.

`GAUDI2_CI=1  RUN_SLOW=1 python -m pytest tests/transformers/tests/models/gpt2/test_modeling_gpt2.py -s -v -k test_contrastive_generate_low_memory`